django/db/backends/oracle/schema.py

```diff
@@ -56,22 +63,35 @@ def alter_field(self, model, old_field, new_field, strict=False):
             # SQLite-ish workaround
             if 'ORA-22858' in description or 'ORA-22859' in description:
                 self._alter_field_type_workaround(model, old_field, new_field)
+            # If we're changing identity column type to non-numeric type
+            # we need to drop identity previous
+            elif 'ORA-30675' in description:
+                self._drop_identity(model._meta.db_table, old_field.column)
+                self.alter_field(model, old_field, new_field, strict)
+            # Modifying identity column isn't supported in Oracle. If we're
+            # changing primary key column to identity column we need to drop
+            # primary key first.
+            elif 'ORA-30673' in description and old_field.primary_key:
+                self._delete_primary_key(model, strict=True)
+                self._alter_field_type_workaround(model, old_field, new_field)
             else:
                 raise
 
     def _alter_field_type_workaround(self, model, old_field, new_field):
         """
         Oracle refuses to change from some type to other type.
         What we need to do instead is:
-        - Add a nullable version of the desired field with a temporary name
-        - Update the table to transfer values from old to new
-        - Drop old column
-        - Rename the new column and possibly drop the nullable property
+        - add a nullable version of the desired field with a temporary name,
+          if new column is a auto field then temporary column can not be
```

Comment:
. If the new column is an auto field, then the temporary column can't be nullable.